pkg/apis/workflow/v1alpha1/types.go

```diff
@@ -252,6 +252,9 @@ type Template struct {
 
 	// Priority to apply to workflow pods.
 	Priority *int32 `json:"priority,omitempty"`
+
+	// ServiceAccount to apply to workflow pods
+	ServiceAccount string `json:"serviceAccount,omitempty"`
```

Comment:
This should be ServiceAccountName instead of ServiceAccount. The latter is deprecated field name of pod specs and is inconsistent with `spec.serviceAcccountName`